rust-mode.el

```diff
@@ -573,6 +584,27 @@ the desired identifiers), but does not match type annotations \"foo::<\"."
             ((not (looking-at (rx (0+ space) "<")))
 	     (throw 'rust-path-font-lock-matcher match))))))))
 
+(defvar rust-string-interpolation-re
+  "\\({[^}]*}\\)"
+  "Regular expression to match interpolation syntax in strings.")
+
+(defun rust-string-interpolation-matcher (limit)
+  (unless (rust-in-cmnt)
+    (save-match-data
+      ;; we presently ignore escaping closing delimiters
```

Comment:
Handling closing delimiters, especially with nesting, is tricky. If you want to handle this, there's an example in groovy-mode: https://github.com/Groovy-Emacs-Modes/groovy-emacs-modes/blob/6713a7d29ba59100db4c168fc6e6bbef277b64ab/groovy-mode.el#L286-L311